### DISCUSSION TITLE
Recommended Way to Load WASM Files from an NPM Package in Next.js?

### DISCUSSION THREAD

**Discussion by bennobuilder** [vercel/next.js#56634] [Help]
### Summary

I'm working with a Next.js project that relies on an `npm` package containing WebAssembly (`WASM`) module/s. The module is bundled using the `rollup-plugin-wasm`. The package structure is as follows:

```
package-name
├── cjs
│   └── [randomid].wasm
└── esm
    └── [randomid].wasm
```

The WASM file IDs are dynamic and change with every build.

I'm facing challenges making these WASM files available as public assets in my Next.js app. I aim to serve these files directly, e.g., via `http://localhost:3000/[randomid].wasm`.

![image](https://github.com/vercel/next.js/assets/57860196/784fd96b-72c5-4902-b95c-9c899fb352ff)

What's the recommended approach for handling this in Next.js? Is copying these files to the `public` directory during the build process the best way, or is there a more idiomatic method in the context of Next.js and `rollup-plugin-wasm`?

Any insights would be greatly appreciated :)

**Ressources:**

`dist` folder structure
![image](https://github.com/vercel/next.js/assets/57860196/9f7aa23c-6bc6-4704-a8d9-c22bab422297)

**Comment by nixjs** [comment]
How to load wasm file without using webpack in nextjs?

**Reply by bennobuilder** [reply]
I couldn't even figure out how to load `wasm` over all. Currently I'm embedding it as `base64` string directly into my library files which are bundled using Rollup and `@rollup/plugin-wasm`.
